operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -375,18 +378,25 @@ func (r *TenantReconciler) updateTnNamespace(ns *v1.Namespace, tnName string) {
 }
 
 // updateTnResQuota updates the tenant resource quota.
-func (r *TenantReconciler) updateTnResQuota(rq *v1.ResourceQuota) {
+func (r *TenantReconciler) updateTnResQuota(ctx context.Context, rq *v1.ResourceQuota, tenantExistingWorkspaces []crownlabsv1alpha1.TenantWorkspaceEntry) error {
 	rq.Labels = r.updateTnResourceCommonLabels(rq.Labels)
 
-	resourceList := make(v1.ResourceList)
+	workspaces := []crownlabsv1alpha1.Workspace{}
 
-	resourceList["limits.cpu"] = *resource.NewQuantity(15, resource.DecimalSI)
-	resourceList["limits.memory"] = *resource.NewQuantity(25*1024*1024*1024, resource.BinarySI)
-	resourceList["requests.cpu"] = *resource.NewQuantity(10, resource.DecimalSI)
-	resourceList["requests.memory"] = *resource.NewQuantity(25*1024*1024*1024, resource.BinarySI)
-	resourceList["count/instances.crownlabs.polito.it"] = *resource.NewQuantity(5, resource.DecimalSI)
+	// get workspaces from tenantWorkspaceEntry
+	for _, tnWs := range tenantExistingWorkspaces {
+		wsLookupKey := types.NamespacedName{Name: tnWs.WorkspaceRef.Name, Namespace: ""}
+		var ws crownlabsv1alpha1.Workspace
```

Comment:
Should propagate the workspaces retrieved previously, and not retrieve them again.